frontend/src/app/shared/icon/icon.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, Input } from '@angular/core';
+
+/**
+ * Wrapper for bootstrap icons
+ *
+ * See https://icons.getbootstrap.com/ to search the list of icons. Use `icon` input to
+ * set the icon name.
+ *
+ * The contents of the icon are hidden for assistive technologies. If the icon is only
+ * decorative, you can use it like this:
+ *
+ * ```html
+ * <button class="btn">
+ *     <lc-icon icon="pencil"></lc-icon>
+ *     Edit
+ * </button>
+ * ```
+ *
+ * If there is no visible text, set the `title` or `aria-label` attribute:
+ *
+ * ```html
+ * <button class="btn">
+ *     <lc-icon icon="pencil" title="edit"></lc-icon>
+ * </button>
+ * ```
+ */
+@Component({
+    selector: 'lc-icon',
+    templateUrl: './icon.component.html',
+})
+export class IconComponent {
+    @Input() icon: string | undefined;
+}
```

Comment:
For documentation purposes and compile-time checks:

```suggestion
    @Input({ required: true }) icon: string | undefined;
```

(I would personally also use `icon!: string;` and get rid of the `undefined` and the `*ngIf` check in the template, since we don't expect this component to be used without an icon, but I know opinions differ on the use of `!`.)